geode-core/src/main/java/org/apache/geode/internal/cache/SerializeNewValueIfNeeded.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache;
+
+import static org.apache.geode.distributed.internal.DistributionConfig.GEMFIRE_PREFIX;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+import org.apache.geode.pdx.internal.InternalPdxInstance;
+
+class SerializeNewValueIfNeeded {
+
+  private static final String ENABLE_PROPERTY = GEMFIRE_PREFIX + "enableSerializeNewValueIfNeeded";
+
+  private final boolean enabled = true || Boolean.getBoolean(ENABLE_PROPERTY);
```

Comment:
This feature needs to be disabled by default before merging to develop. It could potentially be enabled by default in Geode 2.0.

Reply:
Yep!